src/read.c

```diff
@@ -917,14 +917,17 @@ static avifResult avifDecoderItemValidateAV1(const avifDecoderItem * item, avifD
     const avifProperty * av1CProp = avifPropertyArrayFind(&item->properties, "av1C");
     if (!av1CProp) {
         // An av1C box is mandatory in all valid AVIF configurations. Bail out.
-        avifDiagnosticsPrintf(diag, "Item ID %u is missing mandatory av1C property", item->id);
+        avifDiagnosticsPrintf(diag, "Item ID %u of type '%.4s' is missing mandatory av1C property", item->id, (const char *)item->type);
```

Comment:
The declaration of `item->type` is:

```
typedef struct avifDecoderItem
{
    ...
    uint8_t type[4];
    ...
} avifDecoderItem;
```

The use of the `%s` format assumes that the `item->type` array consists of four printable characters, which I believe is the case.

There are no compiler warnings if I simply pass `item->type` without the typecast, but I added the typecast anyway.

Reply:
James: Good point about a corrupted file. I did not consider that. I analyzed the callers of `avifDecoderItemValidateAV1` and found that this function is only called when `item->type` is equal to "av01" or "grid". (This fact is implied by the "ValidateAV1" in the function's name.) So it is safe for this function to print `item->type` using the `%.4s` format.